tensorboard/webapp/notification_center/_views/notification_center_test.ts

```diff
@@ -137,8 +138,58 @@ describe('notification center', () => {
       store.overrideSelector(selectors.getLastReadTime, 1);
       const fixture = TestBed.createComponent(NotificationCenterContainer);
       fixture.detectChanges();
+
       const reddot = fixture.debugElement.query(By.css('.red-dot'));
       expect(reddot).toBeNull();
     });
   });
+  describe('full note link', () => {
+    it('appears when the link is provided', () => {
+      store.overrideSelector(selectors.getNotifications, [
+        {
+          category: CategoryEnum.WHATS_NEW,
+          dateInMs: 1,
+          title: 'test title',
+          content: 'test content',
+          fullNoteLink: 'http://google.com',
```

Comment:
test the invalid URL case including malicious looking ones like: `data:text/html,<script>alert('hi');</script>`.